src/components/try-hcp-callout/components/standalone-link-contents/standalone-link-contents.module.css

```diff
@@ -0,0 +1,31 @@
+.root {
+	align-items: center;
+	color: var(--token-color-foreground-strong);
+	display: flex;
+	font-weight: var(--token-typography-font-weight-medium);
+	gap: 6px;
+	line-height: var(--line-height);
+	margin-bottom: 0;
+	margin-top: 24px;
+
+	& svg {
+		flex-shrink: 0;
+		transition: transform 0.2s;
+
+		@nest :global([class*='card-link']):hover & {
+			transform: translate(2px);
+		}
```

Comment:
Sweet, I'll definitely wrap in the `@media` query!

Re: passing in a class string via props, like "standaloneLink", that seems like a neat approach - but how do we get the provided `className` string into CSS? 🤔 As well, I think we'd still need to the `:global` selector? Admittedly I haven't tried very hard, but it seems harder to do this handoff than it might be for things like CSS custom properties.

The `[class*="card-link"]` selector is definitely a bit brittle - but I figure since `StandaloneLinkContents` isn't abstracted yet, and is just for use in these specific components, that brittleness might be worth the tradeoff and fine for this initial implementation. That being said, open to diving deeper if it feels like the right time.

For near-future, what do you think about [the approach Alex mentioned](https://github.com/hashicorp/dev-portal/pull/1367#discussion_r1025773087), via JS & via a prop (eg maybe `isHovered`)? Intent would be to maybe look at that as part of 🎟 [Assess abstracting atomic "callout" components](https://app.asana.com/0/1202097197789424/1203396252179033/f).

Reply:
✅ updated with the `@media` query!

Let me know what you think about deferring the `[class*="card-link"]` refactor for now 🙇

Reply:
Ah! I didn't even mention where to put this code. 🤦🏻

```css
/* Only enable animation if query is supported and value is no-preference */
@media (prefers-reduced-motion: no-preference) {
  /* `.standaloneLink` would be the class passed to `StandaloneLinkContents` */
  &:hover .standaloneLink svg {
    transform: translate(2px);
  }
}
```

This ☝🏻 would live in `src/components/try-hcp-callout/components/try-hcp-callout/try-hcp-callout.module.css`, and be defined in the `.root` class at the top. Then in the `TryHcpCallout` component, the `s.standaloneLink` would be passed as `className` to the `StandaloneLinkContents` component it renders. No global selector or brittle class selector needed! 😊

Here's a diff of what the changes would be for just `TryHcpCallout`. The compact variant would need the same changes as `TryHcpCallout` is getting. This is working locally for me:

https://github.com/hashicorp/dev-portal/compare/zs.try-hcp-components...amb-zs.try-hcp-components

---

> For near-future, what do you think about https://github.com/hashicorp/dev-portal/pull/1367#discussion_r1025773087, via JS & via a prop (eg maybe isHovered)? Intent would be to maybe look at that as part of 🎟 [Assess abstracting atomic "callout" components](https://app.asana.com/0/1202097197789424/1203396252179033/f).

I'm not sure we should try to change styles via JS except as a last resort, especially since this can be handled with CSS. It keeps the CSS and JS concerns separate to their main uses: CSS for style & presentation and JS for adding functionality.

Reply:
Yes with how the code is currently written, the `translate(2px)` would have to be defined in 2 places:

- `TryHcpCallout`'s `.root` class
- `TryHcpCalloutCompact` `.root` class

As we've chatted about in another thread, these components already have a lot of duplicate code. If defining the CSS twice is now a concern, then we should make `TryHcpCallout` define the main CSS classes, update `TryHcpCalloutCompact` to import that style file and it's own, and apply classes from both files to its markup.

Reply:
I think the question of when & how much we want to duplicate code, vs abstract code here, is a great one 💯

I think there are benefits and drawbacks whichever way we approach this. Personally I'd love to defer further iteration here to a subsequent PR, especially since we have an eye on re-using components like `StandaloneLinkContent` in other components.

Would deferring further iteration here be acceptable?

Reply:
Sure 👍🏻

Reply:
Awesome! I added a note to 🎟 [Assess abstracting atomic "callout" components](https://app.asana.com/0/1202097197789424/1203396252179033/f). For this thread specifically, made a block for `StandaloneLinkContent`. For the other duplicated code here, which I think is mostly related to background styling, made a block for `CardLinkWithBackground` 👍